daily_milestone.go

```diff
@@ -1,29 +1,225 @@
+		if p.Name == "message" {
+			fmt.Println(p.Name)
+		}
+
+		if p.Name == Projectname && p.NameSpace.Path == Namespace {
+			fmt.Println(strconv.Itoa(p.ID))
+			return strconv.Itoa(p.ID), nil
+		}
+	}
+	return "", fmt.Errorf("project %s not found", Projectname)
+}
+
+// It is getting milestones data from the milestone API
+func getMilestones(baseURL string, token string, projectID string) ([]string, error) {
+	project := []milestoneAPI{}
+	list := []string{}
+	strurl := []string{"https://", baseURL, "/api/v4", "/projects/", projectID, "/milestones"}
+	urls := strings.Join(strurl, "")
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", urls, nil)
+	if err != nil {
+		logger.Println(err)
 
-func get_project_id(base_url string, token string, projectname string, namespace string) string {
-	headers := map[string]string{"PrivateToken": token}
-	url := "https://" + base_url + "/projects"
-	page := 1
-	for {
+	}
+	req.Header.Add("PRIVATE-TOKEN", token)
+	resp, err := client.Do(req)
+	if err != nil {
+		logger.Println(err)
+
+	}
+	respByte, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		logger.Println("fail to read response data")
 
 	}
 
+	json.Unmarshal(respByte, &project)
+	defer resp.Body.Close()
+	for _, p := range project {
+		if p.State != "closed" {
+
+			titleList := []string{"Title: ", p.Title, ", DueDate", p.DueDate}
+			y := strings.Join(titleList, " ")
+			list = append(list, y)
+		}
+	}
+
+	return list, nil
 }
 
-//func get_milestones(base_url string, token string, project_id string)[]string{}
-//func create_milestone_data(advance int) []string{}
-//func create_milestones(base_url string, token string, project_id string, milestones string) string{}
+// CreateMilestoneData is used to check the due date using the time package of python
+func createMilestoneData(advance int) []string {
+	today := time.Now().Local()
+	date := today.AddDate(0, 0, advance)
+	list := []string{}
+	/*for i := 0; i < advance; i++ {
+		date := today.AddDate(0, 0, i)                          // To Format to ISOFormat and it converts to string so can be used in list directly
+		datelist := []string{"Title: ", date, "due_date", date} // Was unable to get a map in a list so made this
+		y := strings.Join(datelist, ",")
+		list = append(list, y)
+	}*/
+	fmt.Println(date)
+	fmt.Println(list)
+	return list
+}
+
+func createMilestones(baseURL string, token string, projectID string, milestones []string) string {
+	strurl := []string{"https://", baseURL, "/projects/", projectID, "/milestones"}
+	url := strings.Join(strurl, "")
+	client := &http.Client{}
+	for _, m := range milestones {
+		mbyte := bytes.NewReader([]byte(m))
+		req, err := http.NewRequest("POST", url, mbyte)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		req.Header.Add("PRIVATE-TOKEN", token)
+		resp, err := client.Do(req)
+		if err != nil {
+			logger.Println(err)
+			break
+		}
+		defer resp.Body.Close()
+	}
+	return ("Milestones Created: " + strings.Join(milestones, ""))
```

Comment:
Due to the `break` above you are canceling the whole loop, when one POST request fails.
Failing hard might be ok, but you are always sending back the message, that all milestones got created. That is not true, if an error okkurs. Probably best to check for an error and only give back the message, when no error occurred.
Also sending back an err/nil might be helpful to error check in main.go